Service annotation reconciler (#277)
kubeshop/kusk-gateway: docs/api_autopilot.md

> # API autopilot

Author: @aabedraba please take a look

Kusk gateway supports API auto deployment. 

> Any REST API will typically consist out of a Kubernetes pod that is running the server and a Kubernetes service pointed to it.

Reviewer: ```suggestion
Any REST API will typically consist of a Kubernetes pod(s) that is running the server and a Kubernetes service pointed to it.
```

> ## It just works

Reviewer: I think immediately in this section we should show how to use the annotation rather than putting that in the under the hood section. We can explain in details how it works later but for pure usage, all the info should be here.

Author: I meant it to look like TL;DR thing

Reviewer 2: I agree with Kyle, 

I would add a 
Add this to your service 
```
...
metada: 
  annotation
... 

and apply the change.

doesn't occupy a lot of space and gives a good TLDR visual

> To expose preexisting REST API in the cluster you would just have to execute this:

Reviewer: ```suggestion
To expose a pre-existing REST API in the cluster, you would just have to execute this:
```

```sh
kubectl apply -f svc.yaml
```

> Or edit existing Kubernetes service to add annotation `kusk-gateway/openapi-url` with URL to the location of your OpenAPI

Reviewer: ```suggestion
Or edit an existing Kubernetes service to add the annotation `kusk-gateway/openapi-url` with a URL to the location of your OpenAPI
```

## Under the hood

Let's explain what is going on.

> We added a convenience method that will allow users to easily expose their REST API through Kusk gateway by using `kusk-gateway/openapi-url` annotation, and here is how.

Reviewer: ```suggestion
We added a convenience method that will allow users to easily expose their REST API through Kusk gateway by using the `kusk-gateway/openapi-url` annotation, and here is how.
```

> We added a convenience method that will allow users to easily expose their REST API through Kusk gateway by using the `kusk-gateway/openapi-url` annotation, and here is how.

Reviewer 2: I'd start with "Kusk-gateway has a convenient method..." 
as the docs shouldn't be aware of release timeframe

> Assuming that the user has already set up a pod that is running REST API server code and the pod name is `todo-backend`:

Reviewer: Generally they will set up a Deployment?

Author: I was thinking about the same. I thought here I would go to the lowest possible level which is pods. If you think it is confusing then I don't mind changing it

Reviewer: Please do. The normal flow is to expose a deployment which manages the pods, rather than the pods directly

Author: ```suggestion
Assuming that the user has already set up a pod that is running REST API server code and the deployment name is `todo-backend`:
```
Like this?
